### PR TITLE
feat(medusa,core-flows,types): API to create payment collections for order

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds an API to create a payment collection for order

### CODE REVIEW

File: packages/core/core-flows/src/order/workflows/create-order-payment-collection.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        )
+
+        if (MathBN.lte(pendingPayment, 0)) {
+          throw new MedusaError(
+            MedusaError.Types.NOT_ALLOWED,
+            `Cannot create a payment collection for amount less than 0`
+          )
+        }
+
+        if (input.amount && MathBN.gt(input.amount, pendingPayment)) {
```
Comment (carlos-r-l-rodrigues) [inline packages/core/core-flows/src/order/workflows/create-order-payment-collection.ts:96]: Maybe we don't need this limitation.
one could pre-charge the order 😄

File: packages/core/core-flows/src/order/workflows/create-order-payment-collection.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      list: false,
+    }).config({ name: "payment-collection-query" })
+
+    throwIfActivePaymentCollectionExists({ paymentCollection })
+
+    const paymentCollectionData = transform(
+      { order, input },
+      ({ order, input }) => {
+        const pendingPayment = MathBN.sub(
+          order.summary.current_order_total,
```
Comment (carlos-r-l-rodrigues) [inline packages/core/core-flows/src/order/workflows/create-order-payment-collection.ts]: here the value is serialized. you can read them from `raw_` to keep the precision

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/core/core-flows/src/order/workflows/create-order-payment-collection.ts (hunk 1 above)
Comment (riqwan) [reply packages/core/core-flows/src/order/workflows/create-order-payment-collection.ts:96]: These are just guard rails until we remove them 😄

**Pull request merged**